src/geff/networkx/io.py

```diff
@@ -40,30 +41,39 @@ def get_roi(graph: nx.Graph, position_attr: str) -> tuple[tuple[float, ...], tup
     return tuple(_min.tolist()), tuple(_max.tolist())  # type: ignore
 
 
-def get_node_attrs(graph: nx.Graph) -> list[str]:
-    """Get the attribute keys present on any node in the networkx graph. Does not imply
-    that the attributes are present on all nodes.
+class NXWriterHelper(BaseWriterHelper):
+    """
+    Helper class for writing node or edge attributes to a zarr group.
 
     Args:
-        graph (nx.Graph): A networkx graph
-
-    Returns:
-        list[str]: A list of all unique node attribute keys
+        group: The zarr group to write the attribute to.
+        data_sequence: a sequence of (id, data) pairs. For example, graph.nodes(data=True)
+        position_attr: The name of the position attribute. If None, no position attribute
     """
-    return list({k for n in graph.nodes for k in graph.nodes[n]})
-
 
-def get_edge_attrs(graph: nx.Graph) -> list[str]:
-    """Get the attribute keys present on any edge in the networkx graph. Does not imply
-    that the attributes are present on all edges.
+    def __init__(
+        self,
+        group: zarr.Group,
+        data_sequence: Sequence[tuple[Any, dict[str, Any]]],
+        position_attr: str | None = None,
+    ):
+        super().__init__(group, position_attr)
+        self._data = data_sequence
+        ids = [idx for idx, _ in data_sequence]
+        if len(ids) > 0:
+            self._group["ids"] = np.asarray(ids)
+        elif self._group.name == "/nodes":
```

Comment:
This logic is just to handle if there are no nodes or no edges? Can you add a comment?

Reply:
I added a comment:

```python
        # special corner cases where the graph is empty the ids must still have the right dimension (N,) for nodes, (N, 2) for edges
        elif group.name == "/nodes":
            group["ids"] = np.empty((0,), dtype=np.int64)
        elif group.name == "/edges":
            group["ids"] = np.empty((0, 2), dtype=np.int64)
        else:
 ```